internals/overlord/logstate/opentelemetry/opentelemetry.go

```diff
@@ -0,0 +1,381 @@
+	if attributes == nil {
+		delete(c.resourceAttributes, serviceName)
+		return
+	}
+
+	// Convert attributes to keyValue format.
+	keyValuePairs := make([]keyValue, 0, len(attributes)+1)
+
+	// Add service.name attribute.
+	keyValuePairs = append(keyValuePairs, keyValue{
+		Key:   "service.name",
+		Value: anyValue{StringValue: &serviceName},
+	})
+
+	// Sort other labels to ensure deterministic order.
+	keys := make([]string, 0, len(attributes))
+	for k := range attributes {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		v := attributes[k]
+		keyValuePairs = append(keyValuePairs, keyValue{
+			Key:   k,
+			Value: anyValue{StringValue: &v},
+		})
+	}
+
+	c.resourceAttributes[serviceName] = keyValuePairs
+}
+
+func (c *Client) Add(entry servicelog.Entry) error {
+	if n := len(c.entries); n >= c.options.MaxRequestEntries {
+		// "entries" is full - remove the first element to make room.
+		// Zero the removed element to allow garbage collection.
+		c.entries[0] = entryWithService{}
+		c.entries = c.entries[1:]
+	}
+
+	if len(c.entries) >= cap(c.entries) {
+		// Copy all the elements to the start of the buffer.
+		copy(c.buffer, c.entries)
+
+		// Reset the view into the buffer.
+		c.entries = c.buffer[:len(c.entries):len(c.buffer)]
+
+		// Zero removed elements to allow garbage collection.
+		for i := len(c.entries); i < len(c.buffer); i++ {
+			c.buffer[i] = entryWithService{}
+		}
+	}
+
+	c.entries = append(c.entries, entryWithService{
+		entry:   encodeEntry(entry),
+		service: entry.Service,
+	})
+	return nil
+}
+
+func encodeEntry(entry servicelog.Entry) logRecord {
+	message := strings.TrimSuffix(entry.Message, "\n")
+
+	return logRecord{
+		TimeUnixNano: fmt.Sprintf("%d", entry.Time.UnixNano()),
+		Body:         anyValue{StringValue: &message},
+	}
+}
+
+type payload struct {
+	ResourceLogs []resourceLogs `json:"resourceLogs"`
+}
+
+// Flush sends the buffered logs to the OpenTelemetry collector.
+func (c *Client) Flush(ctx context.Context) error {
+	if len(c.entries) == 0 {
+		return nil // no-op
+	}
+
+	// Group entries by service.
+	serviceBatches := make(map[string][]logRecord)
+	for _, otelEntryWithService := range c.entries {
+		serviceName := otelEntryWithService.service
+		logRecord := otelEntryWithService.entry
+		serviceBatches[serviceName] = append(serviceBatches[serviceName], logRecord)
+	}
+
+	serviceNames := make([]string, 0, len(serviceBatches))
+	for serviceName := range serviceBatches {
+		serviceNames = append(serviceNames, serviceName)
+	}
+	// Sort service names to ensure deterministic order.
+	sort.Strings(serviceNames)
+
+	logs := make([]resourceLogs, 0, len(serviceNames))
+	for _, serviceName := range serviceNames {
+		batch := serviceBatches[serviceName]
+		if len(batch) == 0 {
+			continue
+		}
+
+		resourceAttributes := c.resourceAttributes[serviceName]
+		resource := resource{
+			Attributes: resourceAttributes,
+		}
+		scope := instrumentationScope{Name: "pebble"}
```

Comment:
Similar as with user agent -- let's add `ScopeName string` to `ClientOptions` and pass this in from above and `cmd.ProgramName`, rather than hardcode it here.